src/main/java/org/traccar/protocol/Minifinder2ProtocolDecoder.java

```diff
@@ -248,9 +284,7 @@ protected Object decode(
                     case 0x24:
                         position.setTime(new Date(buf.readUnsignedIntLE() * 1000));
                         long status = buf.readUnsignedIntLE();
-                        if (BitUtil.check(status, 4)) {
-                            position.set(Position.KEY_CHARGE, true);
-                        }
+                        position.set(Position.KEY_CHARGE, BitUtil.check(status, 4));
```

Comment:
Why are you changing this?

Reply:
Bit 4 of the status message always indicates if the device is charging or not. In stead of only adding the `true` state when it's charging, my change passes the actual state along.

Reply:
This is one of the attributes that we only include when it's true, so the original code was correct.